crates/katana/primitives/src/genesis/mod.rs

```diff
@@ -0,0 +1,208 @@
+        let public_key = public_key_from_private_key(private_key);
+        let address = get_contract_address(
+            FieldElement::from(666u32),
+            class_hash,
+            &[public_key],
+            FieldElement::ZERO,
+        );
+
+        (
+            ContractAddress::from(address),
+            Self { private_key, public_key, class_hash, ..Default::default() },
+        )
+    }
+
+    pub fn new_with_balance(
+        private_key: FieldElement,
+        class_hash: ClassHash,
+        balance: U256,
+    ) -> (ContractAddress, Self) {
+        let (address, account) = Self::new(private_key, class_hash);
+        (address, Self { balance, ..account })
+    }
+}
+
+#[serde_with::serde_as]
+#[derive(Debug, serde::Serialize)]
+pub struct FeeTokenConfig {
+    /// The name of the fee token.
+    pub name: String,
+    /// The symbol of the fee token.
+    pub symbol: String,
+    /// The address of the fee token contract.
+    pub address: ContractAddress,
+    /// The decimals of the fee token.
+    pub decimals: u8,
+    // TODO: change to U256
+    /// The total supply of the fee token.
+    pub total_supply: U256,
+    /// The class hash of the fee token contract.
+    #[serde_as(as = "UfeHex")]
+    pub class_hash: ClassHash,
+}
+
+#[serde_with::serde_as]
+#[derive(Debug, serde::Serialize, PartialEq, Eq)]
+pub struct GenesisClass {
+    /// The compiled class hash of the contract class.
+    #[serde_as(as = "UfeHex")]
+    pub compiled_class_hash: CompiledClassHash,
+    /// The casm class definition.
+    #[serde(skip_serializing)]
+    pub casm: CompiledContractClass,
+    /// The sierra class definition.
+    #[serde(skip_serializing)]
+    pub sierra: Option<FlattenedSierraClass>,
+}
+
+#[derive(Debug, serde::Serialize, PartialEq, Eq)]
+pub struct UniversalDeployerConfig {
+    pub class_hash: ClassHash,
+    pub address: ContractAddress,
+}
+
+/// Genesis block configuration.
+#[serde_with::serde_as]
+#[derive(Debug, serde::Serialize)]
+pub struct Genesis {
+    /// The genesis block parent hash.
+    #[serde_as(as = "UfeHex")]
+    pub parent_hash: BlockHash,
+    /// The genesis block state root.
+    #[serde_as(as = "UfeHex")]
+    pub state_root: FieldElement,
+    /// The genesis block number.
+    pub number: BlockNumber,
+    /// The genesis block timestamp.
+    pub timestamp: u128,
+    /// The genesis block L1 gas prices.
+    pub gas_prices: GasPrices,
+    /// The classes to declare in the genesis block.
+    pub classes: HashMap<ClassHash, GenesisClass>,
+    /// The fee token configuration.
+    pub fee_token: FeeTokenConfig,
+    /// The universal deployer (UDC) configuration.
+    pub universal_deployer: Option<UniversalDeployerConfig>,
+    /// The genesis accounts.
+    pub allocations: HashMap<ContractAddress, GenesisAccount>,
+}
+
+/// A helper type for allocating accounts in the genesis block.
+#[must_use]
+pub struct GenesisAllocationsGenerator {
```

Comment:
This is basically a port of https://github.com/dojoengine/dojo/blob/25fbb7fc973cff4ce1273625c4664545d9b088e9/crates/katana/core/src/accounts.rs#L91

Will delete the original one on on the follow up integration PR